### ISSUE TITLE
bug: generated types does not transform import alias paths to relative paths

### ISSUE THREAD

**Issue by AaronDDM** [ionic-team/stencil#3788]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.19.2

### Current Behavior

Running the `build` command and generating `Types` for components does not replace `alias import` paths with relative paths.

Given:
```json
{
  "compilerOptions": {
    "baseUrl": "./",
    "paths": {
      "@interfaces/*": ["./src/interfaces/*"]
    }
  },
}

```

And
```ts
// components/my-component/my-component.tsx
import { MyInterface } from "@interfaces/my-interfaces";
// ...
```

And
```ts
// stencil.config.ts
import { Config } from '@stencil/core';

// https://stenciljs.com/docs/config

export const config: Config = {
  namespace: 'my-components',
  globalStyle: 'src/global/app.css',
  globalScript: 'src/global/app.ts',
  taskQueue: 'async',
  outputTargets: [
    {
      type: 'www',
      // comment the following line to disable service workers in production
      serviceWorker: null,
      baseUrl: 'https://myapp.local/',
    },
    {
      type: "dist",
    }
  ],
};
```

Then output for `dist/types/components/my-component/my-component.d.ts` is:
```ts
// dist/types/components/my-component/my-component.d.ts
import { MyInterface } from '@interfaces/my-interface';
export declare class MyComponent {
  myProperty: MyInterface;
  render(): any;
}
```

### Expected Behavior

The generated `dist/types/*` files should transform aliased import paths automatically to relative paths, otherwise the generates types are invalid.

Given:
```json
{
  "compilerOptions": {
    "baseUrl": "./",
    "paths": {
      "@interfaces/*": ["./src/interfaces/*"]
    }
  },
}

```

And
```ts
// components/my-component/my-component.tsx
import { MyInterface } from "@interfaces/my-interfaces";
// ...
```

And
```ts
// stencil.config.ts
import { Config } from '@stencil/core';

// https://stenciljs.com/docs/config

export const config: Config = {
  namespace: 'my-components',
  globalStyle: 'src/global/app.css',
  globalScript: 'src/global/app.ts',
  taskQueue: 'async',
  outputTargets: [
    {
      type: 'www',
      // comment the following line to disable service workers in production
      serviceWorker: null,
      baseUrl: 'https://myapp.local/',
    },
    {
      type: "dist",
    }
  ],
};
```

Then output for `dist/types/components/my-component/my-component.d.ts` is:
```ts
// dist/types/components/my-component/my-component.d.ts
import { MyInterface } from '../../interfaces/my-interface';
export declare class MyComponent {
  myProperty: MyInterface;
  render(): any;
}
```

### Steps to Reproduce

1. In your `tsconfig.json` define an alias import under `compilerOptions.paths` (e.g. `"@interface/*": ["./src/interface/*"],`)
2. Use the alias path to import a file using the alias path in any of your source files (e.g. `import { MyInterface } from "@interface/my-interfaces";`)
3. Run `stencil build`

### Code Reproduction URL

https://github.com/AaronDDM/stencil-type-alias-bug-repo

**Comment by tanner-reits** [comment]
Hey @AaronDDM 👋 
Thanks for reporting this issue! I was able to hop in the reproduction URL and verify the problem with the generated type definitions. I'm gonna get this labeled so it'll be pulled into our backlog to be resolved in the future!

**Comment by MooDySmiles** [comment]
Is there any workaround for this in the meantime?

**Comment by rwaskiewicz** [comment]
@AaronDDM The fix for this issue has been released in Stencil v3.1.0. I apologize this notice didn't go out as a part of the release - it managed to slip through the cracks. In testing this against the provided reproduction case, I've validated the output for `dist/types/components/my-component/my-component.d.ts is` is set as expected.

@vcantor can you please do me a favor and open a new issue with a minimal reproduction case for the team to look at? The reproduction provided by the original poster of the issue appears to be working correctly using v3.2.0 and `transformAliasedImportPaths` - we'd need some more information/a reproduction to investigate futher.

Thanks all!
